charts/karpenter/templates/provisioning.karpenter.sh_provisioners.yaml

```diff
@@ -88,10 +100,14 @@ spec:
                   - key
                   type: object
                 type: array
-              ttlSeconds:
-                description: TTLSeconds determines how long to wait before attempting to terminate a node.
-                format: int32
-                type: integer
+              utilization:
```

Comment:
Agree -- thoughts on a better name?

Reply:
Just brainstorming ... maybe if `expiration` were called `maximumNodeLifetime` then this would be `minimumNodeUtilization`?

Reply:
Thinking on this further, I had hoped to expand this section to include all defrag related concepts. You can think of Karpenter as a set of decoupled controllers. The "expiration controller" is pretty simple and straight forward. The "utilization controller" is quite a bit more complex, and is responsible for maximizing the utilization of the cluster. I could imagine potentially breaking this thing apart, but I'm not sure yet. Further, the utilization.ttlSeconds field could be used for defrag (i.e., don't defrag until a certain amount of time has elapsed).

Reply:
I see, I think it's a bit hard (for me at least) to envision what this will eventually look like. Today there's a simple binary definition of underutilized (no non-DaemonSet Pods), I assume once we support actually saying what 'utilization' is an acceptable floor, we'll have to get into:

1. wiring up metrics to the controller?
2. specifying which resources should be utilized how much (disk, network, memory, cpu, other)?

Reply:
Not to make things more complicated, but I'm also wondering if Provisioner will eventually become more like (say) a Deployment, just like how Deployments create ReplicaSets, etc (sorry if I have that wrong, hopefully the gist is clear), the Provisioner might create a NodeUtilizationMonger controller (obviously with a better name), NodeGrimReaper, etc. So then we'd want to think about what the spec for those controllers would look like?

Reply:
Yeah totally agree. I don't know what that will look like either in the fullness of time. One way forward is to just make our best guess w/ current information and keep our APIs alpha as long as we can, so we can backwards compat these things. I'd expect our API would need to stabilize by v1.0.

One nice thing with this whole question is that because the provisioner "creates, but doesn't own" the nodes, it means that it's fairly straightforward to uninstall/reinstall without taking an outage. The nodes will simply drift and by adopted in the meantime.

Reply:
By this argument, I think it makes sense to bring these fields up a level to top level fields like:
```
maximumNodeLifetimeSeconds
underutilizationTTLSeconds
```
The semantic will be, "if undefined, don't trigger behavior". This will enable users to turn off expiry/underutilization.

WDYT?

Reply:
Makes sense to me